kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -102,11 +101,28 @@ public TransactionBuilder withPartitionColumns(Engine engine, List<String> parti
     return this;
   }
 
+  /**
+   * There are three possible cases when handling clustering columns via `withClusteringColumns`:
+   *
+   * <ul>
+   *   <li>1. Clustering columns are not set (i.e., `withClusteringColumns` is not called):
+   *   <li>- No changes are made related to clustering.
+   *   <li>- For table creation, the table is initialized as a non-clustered table.
+   *   <li>- For table updates, the existing clustered or non-clustered state remains unchanged
+   *   <li>(i.e., no protocol or domain metadata updates).
+   *   <li>2. Clustering columns are an empty list:
+   *   <li>- This is equivalent to executing `ALTER TABLE ... CLUSTER BY NONE` in Delta.
+   *   <li>- The table remains a clustered table, but its clustering domain metadata is updated
+   *   <li>to reflect an empty list of clustering columns.
+   *   <li>3. Clustering columns are a non-empty list:
+   *   <li>- The table is treated as a clustered table.
+   *   <li>- We update the protocol (if needed) to include clustering writer support and set the
+   *   <li>clustering domain metadata accordingly.
+   * </ul>
+   */
   @Override
   public TransactionBuilder withClusteringColumns(Engine engine, List<Column> clusteringColumns) {
-    if (!clusteringColumns.isEmpty()) {
-      this.clusteringColumns = Optional.of(clusteringColumns);
-    }
+    this.clusteringColumns = Optional.of(clusteringColumns);
```

Comment:
It is a bit confusing API, but it is because of Kernel not having intent based APIs (nothing due to this Pr). Ideally by having `withClusteringRemoved` as an API would be clear, instead of overloading this API.

We will take this account when designing the intent based txn start APIs